scheduler/scheduling/evaluator/evaluator_network_topology.go

```diff
@@ -0,0 +1,233 @@
+/*
+ *     Copyright 2024 The Dragonfly Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package evaluator
+
+import (
+	"sort"
+	"strings"
+	"time"
+
+	"d7y.io/dragonfly/v2/pkg/math"
+	"d7y.io/dragonfly/v2/pkg/types"
+	"d7y.io/dragonfly/v2/scheduler/networktopology"
+	"d7y.io/dragonfly/v2/scheduler/resource"
+)
+
+const (
+	// Finished piece weight.
+	networkTopologyFinishedPieceWeight float64 = 0.2
+
+	// Parent's host upload success weight.
+	networkTopologyParentHostUploadSuccessWeight = 0.2
+
+	// Free upload weight.
+	networkTopologyFreeUploadWeight = 0.15
+
+	// Network topology weight.
+	networkTopologyNetworkTopologyWeight = 0.12
+
+	// Host type weight.
+	networkTopologyHostTypeWeight = 0.11
+
+	// IDC affinity weight.
+	networkTopologyIDCAffinityWeight = 0.11
+
+	// Location affinity weight.
+	networkTopologyLocationAffinityWeight = 0.11
+)
+
+const (
+	// Maximum score.
+	networkTopologyMaxScore float64 = 1
```

Comment:
Move `maxScore` and `minScore` from `evaluator_base.go` to `evaluator.go` and use common `maxScore` and `minScore`.